pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -397,8 +397,9 @@ def reduce(self, tensor, group: Optional[Any] = None, reduce_op: Union[ReduceOp,
             tensor = sync_ddp_if_available(tensor, group, reduce_op=reduce_op)
         return tensor
 
-    def training_step(self, *args, **kwargs) -> Optional[Any]:
-        return self.model(*args, **kwargs)
+    def training_step(self, step_kwargs: Dict[str, Union[Any, int]]) -> STEP_OUTPUT:
```

Comment:
Can we stick to `*args` and `**kwargs` please? We can rename them to `step_args` and `step_kwargs`, but I think it's important that we have positional and keyword arguments, as this is the only way to possibly deal with [PEP 570](https://www.python.org/dev/peps/pep-0570/) and [PEP 3102](https://www.python.org/dev/peps/pep-3102/)

Reply:
@justusschock @awaelchli do you recall why the accelerator had a different signature/convention compared to the training type plugin?

Reply:
@justusschock Agree that *args, **kwargs is better, and it's align with what torch model training step will take and it's more flexible. I'm copying Accelerator's signature in this PR because all call to training_type_plugin.training_step was called through accelerator.training_step() before. And Accelerator's signature is 
https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/accelerators/accelerator.py#L124-L130
If we use `*args, **kwargs` now will loose the requirement for the typing in a way, which I was concerned that could cause unexpected failures.

Reply:
@four4fish I get it. I think we still should move on with args and kwargs :)

@ananthsub yes I think that should be part of the loops. So that plugins are unrelated to that.

Reply:
@justusschock sounds good, will update and mark as breaking change

Reply:
Issue created, will address the issue before update this PR
https://github.com/PyTorchLightning/pytorch-lightning/issues/10907